python/primihub/FL/model/logistic_regression/homo_lr_dev.py

```diff
@@ -299,27 +301,50 @@ def broadcast_global_model_param(self):
         self.host_channel.send("global_param", self.theta)
         self.guest_channel.send("global_param", self.theta)
 
-    def loss_and_acc(self, losses, accs, weights):
-        return np.average(losses, weights=weights), np.average(accs, weights=weights)
-
-    def metrics_log(self, weights):
+    def metrics_log(self, num_examples_weights,
+                          num_positive_examples_weights,
+                          num_negtive_examples_weights):
+        # loss
         penalty_loss = 0.5 * self.alpha * self.theta.dot(self.theta)
         host_loss = self.host_channel.recv("host_loss") - penalty_loss
         guest_loss = self.guest_channel.recv("guest_loss") - penalty_loss
         losses = [host_loss, guest_loss]
-
+        loss = np.average(losses, weights=num_examples_weights) + penalty_loss
+        self.loss_value = loss # used for check convergence
+        
+        # acc
         host_acc = self.host_channel.recv("host_acc")
         guest_acc = self.guest_channel.recv("guest_acc")
         accs = [host_acc, guest_acc]
+        acc = np.average(accs, weights=num_examples_weights)
+
+        # fpr, tpr
+        host_fpr = self.host_channel.recv("host_fpr")
+        guest_fpr = self.guest_channel.recv("guest_fpr")
+        host_tpr = self.host_channel.recv("host_tpr")
+        guest_tpr = self.guest_channel.recv("guest_tpr")
+        host_thresholds = self.host_channel.recv("host_thresholds")
+        guest_thresholds = self.guest_channel.recv("guest_thresholds")
+        fpr, tpr, thresholds = fpr_tpr_merge2(host_fpr, host_tpr, host_thresholds,
```

Comment:
whether it has the problem of information leakage?

Reply:
Yes, the fpr, tpr, and thresholds arrarys do have privacy issues, but here we assume the arbiter is a trusted third party.